recipe/thirdparty/providers/facebook.go

```diff
@@ -80,18 +80,18 @@ func Facebook(config tpmodels.FacebookConfig) tpmodels.TypeProvider {
 					}
 					userInfo := response.(map[string]interface{})
 					ID := userInfo["id"].(string)
-					email := userInfo["email"].(string)
-					if email == "" {
+					email, emailOk := userInfo["email"].(string)
+					if !emailOk {
 						return tpmodels.UserInfo{
 							ID: ID,
 						}, nil
 					}
-					isVerified := userInfo["verified_email"].(bool)
+					isVerified, isVerifiedOk := userInfo["verified_email"].(bool)
 					return tpmodels.UserInfo{
 						ID: ID,
 						Email: &tpmodels.EmailStruct{
 							ID:         email,
-							IsVerified: isVerified,
+							IsVerified: isVerified && isVerifiedOk,
```

Comment:
Should you do isVerifiedOK && isVerified?

Reply:
shouldn't matter, both are booleans